src/Symfony/Component/Mailer/Bridge/Amazon/Credential/InstanceCredentialProvider.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\Bridge\Amazon\Credential;
+
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Component\Mailer\Exception\RuntimeException;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * Based on: aws-sdk-php / Credentials/InstanceProfileProvider.php.
+ *
+ * @author Karoly Gossler <connor@connor.hu>
+ */
+class InstanceCredentialProvider
+{
+    const SERVER_URI_TEMPLATE = 'http://169.254.169.254/latest/meta-data/iam/security-credentials/%role_name%';
+
+    public function __construct(HttpClientInterface $client = null, int $retries = 3)
+    {
+        $this->retries = $retries;
+
+        if (null === $client) {
+            if (!class_exists(HttpClient::class)) {
+                throw new LogicException(sprintf('You cannot use "%s" as the HttpClient component is not installed. Try running "composer require symfony/http-client".', __CLASS__));
+            }
+
+            $this->client = HttpClient::create();
+        } else {
+            $this->client = $client;
```

Comment:
Nitpicking mode on: no need for an else here. Simply do `$this->client = $client` above where you assign retries. Then check if `$this->client` is null instead of `$client` and you can remove the if.

Reply:
@mdeboer I can remove the else, but you're right.

Reply:
Sorry yeah, I meant the else hehe 😜 Blame the heatwave..